Python_Scripts/script_regroup_models.py

```diff
@@ -0,0 +1,337 @@
+"""
+Imports the SED-ML models and the SBML models from Biomodels and regroups them
+according to model properties (state variables, reactions, model name)
+in a newly created folder structure.
+One folder then corresponds to one benchmark model, with possibly multiple SBML
+files belonging to one benchmark model.
+"""
+
+# Note: libsedml must be imported before libsbml for whatever reason
+
+import libsedml
+import libsbml
+import os
+import shutil
+import pandas as pd
+import re
+
+from C import (DIR_MODELS_SEDML, DIR_MODELS_REGROUPED, DIR_MODELS, DIR_BASE)
+from setTime_BioModels import timePointsBioModels
+
+
+# get all models
+def regroup_models():
+    # get all models, list information about them
+    list_directory_sedml = sorted(os.listdir(DIR_MODELS_SEDML))
+    model_info = []
+
+    for sedml_model in list_directory_sedml:
+        # get the model name and year
+        model_name, model_year = _parse_model_name(sedml_model)
+        if model_name is None or model_year is None:
+            continue
+
+        # get the paths
+        sedml_folder = os.path.join(DIR_MODELS_SEDML, sedml_model)
+        sedml_file = os.path.join(sedml_folder, f'{sedml_model}.sedml')
+
+        sbml_folder = os.path.join(sedml_folder, 'sbml_models')
+        sbml_files = [os.path.join(sbml_folder, sbml_model)
+                      for sbml_model in sorted(os.listdir(sbml_folder))]
+
+        if os.path.exists(sedml_file):
+            model_info = _check_sedml_submodels(sedml_file, sbml_files,
+                                                model_name, model_year,
+                                                model_info)
+        else:
+            # only one SBML file, no SED-ML info. A benchmark model on its own.
+            sbml_path = os.path.abspath(sbml_files[0])
+            model_info = _check_biomodels_model(sedml_model, sbml_path,
+                                                model_name, model_year,
+                                                model_info)
+
+    # Group sbml models and give them short Identifiers (e.g., Bachmann2011a)
+    model_info_df = _group_models_by_id(model_info)
+
+    return model_info_df
+
+
+def _group_models_by_id(model_info):
+    # We've run through all models. Now, let's generate short identifier for
+    # the groups of models, which should belong together
+    known_ids = {}
+    for model in model_info:
+        if model['long_id'] not in known_ids:
```

Comment:
doc: if the model long id is not the same as a previous one, create a new short id